### ISSUE TITLE
Idea Discussion: Support High-Level System Groups in Toolbox

### ISSUE THREAD

**Issue by DillonJ** [Spine-project/Spine-Toolbox#1553]
Similar to group functionality, this "system-group" idea would allow us to create groups containing all objects classes. The idea would be to create containers for sub-systems - at least this is one use-case.
Desired features 
 - We might want to define what kinds of objects classes a group can contain. Maybe this is an extension/generalisation of existing group functionality?
 - Assign and view members of a system-group - similar to existing group objects
 - Support in graph-view where by default a system group is displayed as as a single icon but can be expanded to show it's members
 - When collapsed (shown as a single icon) only the relationships with external elements will be shown
 - Maybe a system group could contain system groups and we could have nesting behaviour.
 - Option to expand system group in a separate graph window

Thoughts?

**Comment by jkiviluo** [comment]
It would be great as long as it remains understandable and clearly separate functionality from the current inside-class group. Maybe another name than 'group'? Or then re-think the names of both current group and this high-level group?

It would be also nice if you could then use these groups to include (or exclude) them from a scenario.

**Comment by DillonJ** [comment]
Yeah. I was thinking this could work really nicely in a number of ways, particularly, inside a SpineOpt model. A system could be an individual power plant with different components, or it could be an entire system. For example, you could represent countries as one of these system group objects and drill down to see the individual elements, nodes, units etc. You could also use them to control things like temporal resolution, stochastic structure etc. that the lower level entities would inherit. 

Could be really useful

**Comment by jkiviluo** [comment]
Added bags to the new DB schema. Bags could include entities of any type while groups would continue to include entities only from the same class.
